client/layout/sidebar/expandable.jsx

```diff
@@ -87,7 +87,7 @@ export const ExpandableSidebarMenu = ( {
 	};
 
 	const onLeave = () => {
-		if ( expanded || isTouch || ! config.isEnabled( 'nav-unification' ) ) {
+		if ( isTouch || ! config.isEnabled( 'nav-unification' ) ) {
```

Comment:
I wonder if a small comment here could help explain why the `onLeave` handler is necessary and why the behaviour needs to change for touch devices?

Just some context so that folks don't accidentally re-introduce a bug later. These issues are hard to spot and we don't have any test coverage 😢